### DISCUSSION TITLE
What does this solution do `OpenTelemetry.AutoInstrumentation.Native`?

### DISCUSSION THREAD

**Discussion by varkey98** [open-telemetry/opentelemetry-dotnet-instrumentation#3183] [Q&A] [answered]
I want to undestand how we get the loggingbuilder here: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/src/OpenTelemetry.AutoInstrumentation/Instrumentations/Logger/LoggingBuilderIntegration.cs? Can I generalise this and implement a similar class for getting WebApplicationBuilder whenever it gets called?

**Comment by Kielek** [comment]
I think that you should start with reading this document https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/design.md#bytecode-instrumentation

Especially "Bytecode instrumentation" section.

What is your use case? What do you want to do with WebApplicationBuilder?

**Reply by varkey98** [reply]
Thanks, just figured that out. What I want to do is to add a middleware to an asp net app. I think that I can do this with a hosting startup assembly, but wanted to understand if this was an option as well

**Reply by varkey98** [reply]
@Kielek do you think what I'm trying to do is possible?

**Reply by Kielek** [reply, marked as answer]
Sorry. I do not understand the use case. Could you please be more descriptive?

In general with monkey-patching aka bytecode changes you can do anything with the code, but is it not for free. If possible you should avoid it as it reduce performance/optimisations.

**Reply by varkey98** [reply]
```
var builder = WebApplication.CreateBuilder(args);
var app = builder.Build();
```

@Kielek  If I have an app which is being built like this, I'd like to get this builder instance and do something `builder.Services.Add` before `builder.build()` gets called.
